ci/test_narwhals.sh

```diff
@@ -15,8 +15,20 @@ trap "EXITCODE=1" ERR
 set +e
 
 rapids-logger "pytest narwhals"
-git clone https://github.com/narwhals-dev/narwhals --depth=1 -b stable
```

Comment:
`-b` I think still accepts tags

```bash
git clone https://github.com/narwhals-dev/narwhals --depth=1 -b "v${NARWHALS_VERSION}"
```

Reply:
Ok I'm curious if folks have a preference. This is what we do for [pandas](https://github.com/rapidsai/cudf/blob/branch-25.06/python/cudf/cudf/pandas/scripts/run-pandas-tests.sh#L34-36)

Reply:
Ah gotcha. I think that pandas script is a bit different because it was also used a lot for local development (which was then also pulled into a CI script) which maybe why it's configured that way?

Using something simpler here would be nice, but I also don't have a strong preference.

Reply:
I think the suggestion above is a good one. Let's directly clone the tag we want, since it is known ahead of time.